flyteadmin/pkg/manager/impl/execution_manager_test.go

```diff
@@ -4088,6 +4088,290 @@ func TestListExecutions_LegacyModel(t *testing.T) {
+
+			lpClosure := admin.LaunchPlanClosure{
+				ExpectedInputs: expectedLPInputs,
+			}
+			lpClosureBytes, err := proto.Marshal(&lpClosure)
+			assert.NoError(t, err)
+
+			getLaunchPlanCb := func(id interfaces.Identifier) (models.LaunchPlan, error) {
+				assert.Equal(t, launchPlanID.GetProject(), id.Project)
+				assert.Equal(t, launchPlanID.GetDomain(), id.Domain)
+				assert.Equal(t, launchPlanID.GetName(), id.Name)
+				assert.Equal(t, launchPlanID.GetVersion(), id.Version)
+				activeState := int32(admin.LaunchPlanState_ACTIVE)
+				return models.LaunchPlan{
+					LaunchPlanKey: models.LaunchPlanKey{
+						Project: id.Project,
+						Domain:  id.Domain,
+						Name:    id.Name,
+						Version: id.Version,
+					},
+					WorkflowID: 1, // Assuming a valid workflow ID
+					Spec:       launchPlanSpecBytes,
+					State:      &activeState,
+					Closure:    lpClosureBytes,
+				}, nil
+			}
+			mockRepo.LaunchPlanRepo().(*repositoryMocks.MockLaunchPlanRepo).SetGetCallback(getLaunchPlanCb)
+
+			getWorkflowCb := func(id interfaces.Identifier) (models.Workflow, error) {
+				return models.Workflow{
+					WorkflowKey:             models.WorkflowKey{Project: workflowID.GetProject(), Domain: workflowID.GetDomain(), Name: workflowID.GetName(), Version: workflowID.GetVersion()},
+					RemoteClosureIdentifier: remoteClosureIdentifier, // Use the global const
+				}, nil
+			}
+			mockRepo.WorkflowRepo().(*repositoryMocks.MockWorkflowRepo).SetGetCallback(getWorkflowCb)
+
+			if tt.concurrencyPolicy != nil {
+				mockRepo.ExecutionRepo().(*repositoryMocks.MockExecutionRepo).SetCountCallback(
+					func(ctx context.Context, input interfaces.CountResourceInput) (int64, error) {
+						t.Logf("Received %d filters in SetCountCallback for test: %s", len(input.InlineFilters), tt.name)
+						for i, f := range input.InlineFilters {
+							gexpr, _ := f.GetGormQueryExpr()
+							t.Logf("Filter %d: Entity=%v, Field=%s, Args=%v", i, f.GetEntity(), f.GetField(), gexpr.Args)
+						}
+
+						var foundProject, foundDomain, foundLpName, foundPhase bool
+
+						for _, filter := range input.InlineFilters {
+							gormQueryExpr, err := filter.GetGormQueryExpr()
+							if !assert.NoError(t, err) {
+								continue
+							}
+							if !assert.NotNil(t, gormQueryExpr.Args) {
+								continue
+							}
+
+							entity := filter.GetEntity()
+							field := filter.GetField()
+
+							if entity == common.Execution {
+								if field == "execution_project" {
+									if val, ok := gormQueryExpr.Args.(string); ok && val == launchPlanID.GetProject() {
+										foundProject = true
+									}
+								} else if field == "execution_domain" {
+									if val, ok := gormQueryExpr.Args.(string); ok && val == launchPlanID.GetDomain() {
+										foundDomain = true
+									}
+								} else if field == "phase" {
+									if phaseList, ok := gormQueryExpr.Args.([]string); ok && len(phaseList) > 0 {
```

Comment:
Can this be updated to exact match the length of the phase list? Would make the unit test more robust to catching changes.